atlp-rwanda/atlp-devpulse-bn
# Ft applicant dashboard

### PR Description
This PR refactors the user model and updates some files necessary for applicant dashboard cards on the frontend

### Description of tasks that were expected to be completed

- Refactor user, cohort and trainee model and resolver functions
- Create models for attendance and performance and their respective resolver functions and graphql schemas


### How has this been tested?
- Clone repo
- Checkout branch `ft-applicant-dashboard`
- Make sure that you set the environment variables referring to `.env.example`
- Run `npm install`, then `npm run dev` to check if the app is running successfully
- Navigate to `localhost:5000` and query the server

- `mutation UpdateApplicationPhase` to update to trainee pass in these variables 
```
{  "userId": "66fdbd3d1b3a0954151e2a01",
  "newPhase": "Accepted",
  "cohortId": "66e9f4bc1bf835a4d95cfffb",
  
}
```
- Use query `GetTraineeAttendance` and query `GetTraineePerformance` to get user's attendance and performance

### Track PR (issue number & link)
The issue link is the on front end click [here](https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/150)

### Screenshots (If appropriate especially on the frontend)
N/A
